internal/manifests/manifests.go

```diff
@@ -65,6 +67,19 @@ func (m *Manifests) CreateClusterManifest(ctx context.Context, params operations
 			fileName, cluster.ID, *params.CreateManifestParams.Content)
 		return common.GenerateErrorResponderWithDefault(errors.New("failed to base64-decode cluster manifest content"), http.StatusBadRequest)
 	}
+	extension := filepath.Ext(fileName)
+	if extension == ".yaml" || extension == ".yml" {
+		var s map[interface{}]interface{}
+		if yaml.Unmarshal(manifestContent, &s) != nil {
```

Comment:
Isn't there a Validation function to be used?
Would it always work with this map type?